types/src/transaction/mod.rs

```diff
@@ -453,6 +455,8 @@ impl WriteSetPayload {
     }
 }
 
+pub type DiemSignedTransaction = SignedTransaction<DiemTransactionAuthenticator>;
```

Comment:
what is client code may see? A DiemSignedTransaction or SignedTransaction?

Current transaction builder generates type code for client applications, used name SignedTransaction.
If this change will cause generated client code to have DiemSignedTransaction, I think it's a bad idea to rename it.

Reply:
More specifically, SignedTransaction is the part of interface to client applications, we should not break it.